content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -449,6 +449,179 @@ data class CreateBoardingTicketInitiator @JsonConstructor constructor(private va
 }
 ```
 
+## Write the `GiftVoucherToFriend` flow
+
+In Corda, you can transfer a state to someone else. The `GiftVoucherToFriend` flow lets you transfer the ownership of the issued voucher to another person.
+
+Now that you've written the `CreateAndIssueMarsVoucher` and `CreateBoardingTicket` flows, try writing the `GiftVoucherToFriend` flow.
+
+You will need these variables:
+
+* `voucherID`
+* `holder`
+
+You must inject these services:
+
+* `FlowEngine`: Provides basic control over a flow it is injected into, such as calling subflows, performing asynchronous tasks and putting the flow to sleep for a period of time. This flow needs `FlowEngine` because you are utilizing subflows such as `FinalityFlow`.
+* `FlowIdentity`: Obtains the identity of the node running the flow. You must inject this service because one of the output states needs its own identity.
+* `FlowMessaging`: Used for creating and closing flow sessions as well as sending and receiving data between flows via a flow session. Use this service to create flow sessions that allow communication between counterparties.
+* `TransactionBuilderFactory`: Used for constructing, verifying and signing the transactions. In this case, it builds a transaction.
+* `IdentityService`: Provides methods to retrieve Party and AnonymousParty instances. You need it because your flow involves counterparties.
+* `NotaryLookupService`: Finds information on notaries in the network. Add it because your transaction requires a notary.
+* `JsonMarshallingService`: Parses arbitrary content in and out of JSON using standard, approved mappers. You need it because in Corda 5 all flow parameters are in the JSON format.
+* `PersistenceService`: Provides an API for interacting with the database. It has functions mirroring Java’s EntityManager for working with entities. Also, it provides functions for executing predefined named queries and polling for results. It hides the complexity of asynchronously interacting with the database which, in a high-availability environment, could be running on a separate process.
+
+{{< note >}}
+Similarly to the `CreateAndIssueMarsVoucher` flow, this flows needs a responder flow that responds to the request to update the ledger. Also, you need to query the `MarsVoucher` and the `BoardingTicket`.
```

Comment:
I might break this out into two separate lines, not include it as a note. Like the services, this info is essential for the user to know. 

Maybe link back to the section where they wrote the queries for `RedeemBoardingTicketWithVoucher`, maybe something like:
```suggestion
Like the `CreateAndIssueMarsVoucher` flow, the `GiftVoucherToFriend` flow needs a [responder flow](#write-the-responder-flow) that responds to the request to update the ledger. 

You also need to query the `MarsVoucher` and the `BoardingTicket`. Refer to the section on [implementing queries](#implement-queries) for the `RedeemBoardingTicketWithVoucher` for guidance.
```